### DISCUSSION TITLE
Matlab Support availability

### DISCUSSION THREAD

**Discussion by CsFreiber** [microchip-pic-avr-solutions/avr-iot-cellular-arduino-library#23] [Q&A] [answered]
Hello Simen,

I was curious and looked up the supported hardware in **Matlab for Embedded Coder** and so I found out, that the AVR 8 Bit MCUs are supperted (however under the name from ATmel, which indicates, that these are "old" MCUs https://de.mathworks.com/hardware-support/atmel.html?s_tid=srchtitle_hardware_catalog_main_1_AVR ) We are currently programming with your libraries, from DxCore and also with ohter including our own made in VS Code. It works really fine and we are content with the results, but out of curiosity I would test if it is easier or quicker to programm in Matlab/Simulink, due to the model based softver development. So my question is following: is there a Matlab support also for the **AVR128DB48** or even for the **Sequans Monarch 2** modem? Have someone already tried this out?
I am looking forward for your answer and opinion and thank you for your previous support!

Best regards

CsFreiber

**Comment by M65649** [comment, marked as answer]
Hello @CsFreiber,

That's not something that we've explored (or that I've heard other customers explore) so far on this kit. I see that there's a Microchip package as well for PIC and SAM, but not for the newer AVRs. I'm uncertain of how it would work in collaboration with the modem. I guess you could try to make your model in e.g., simulink and export it to an older AVR and make modifications manually (if needed) to port it to the DB.

**Reply by CsFreiber** [reply]
Hello Simen,

thank you for your quick response! Yeah, I have also found this package. Maybe it is worth a try as you suggest, if we are having an even more complex system. But I think at the moment we stay at this setup.

Best regards

CsFreiber
